platform/app/src/appInit.js

```diff
@@ -111,7 +111,7 @@ async function appInit(appConfigOrFunc, defaultExtensions, defaultModes) {
           ? appConfig.modesConfiguration[id]
           : {};
 
-      mode = mode.modeFactory({ modeConfiguration });
+      mode = await mode.modeFactory({ modeConfiguration, loadModules });
```

Comment:
And the modeFactory is made async capable so that it can wait for files to load/modules internal to load.